x/docs/internal/keeper/keeper_test.go

```diff
@@ -190,125 +190,173 @@ func TestKeeper_ShareDocument_EmptyLists(t *testing.T) {
 		Metadata:   TestingDocument.Metadata,
 		Checksum:   TestingDocument.Checksum,
 	}
-	k.ShareDocument(ctx, newDocument)
+	err := k.ShareDocument(ctx, TestingSender, ctypes.Addresses{newRecipient}, newDocument)
+	assert.Nil(t, err)
 
-	sentDocsBz := store.Get(k.getSentDocumentsStoreKey(TestingDocument.Sender))
-	receivedDocsBz := store.Get(k.getReceivedDocumentsStoreKey(TestingDocument.Recipient))
+	sentDocsBz := store.Get(k.getSentDocumentsStoreKey(TestingSender))
+	receivedDocsBz := store.Get(k.getReceivedDocumentsStoreKey(TestingRecipient))
 	newReceivedDocsBz := store.Get(k.getReceivedDocumentsStoreKey(newRecipient))
 
-	var sentDocs, receivedDocs, newReceivedDocs types.Documents
+	var sentDocs, receivedDocs, newReceivedDocs types.DocumentIds
 	cdc.MustUnmarshalBinaryBare(sentDocsBz, &sentDocs)
 	cdc.MustUnmarshalBinaryBare(receivedDocsBz, &receivedDocs)
 	cdc.MustUnmarshalBinaryBare(newReceivedDocsBz, &newReceivedDocs)
 
-	assert.Equal(t, 2, len(sentDocs))
-	assert.Contains(t, sentDocs, TestingDocument)
-	assert.Contains(t, sentDocs, newDocument)
+	assert.Equal(t, 1, len(sentDocs))
+	assert.Contains(t, sentDocs, TestingDocument.Uuid)
 
 	assert.Equal(t, 1, len(receivedDocs))
-	assert.Contains(t, receivedDocs, TestingDocument)
+	assert.Contains(t, receivedDocs, TestingDocument.Uuid)
 
 	assert.Equal(t, 1, len(newReceivedDocs))
-	assert.Contains(t, newReceivedDocs, newDocument)
+	assert.Contains(t, newReceivedDocs, newDocument.Uuid)
+}
+
+func TestKeeper_ShareDocument_SameDocumentDifferentUuid(t *testing.T) {
+	cdc, ctx, k := SetupTestInput()
+	documentsIds := types.DocumentIds{TestingDocument.Uuid}
+
+	store := ctx.KVStore(k.StoreKey)
+	store.Set(k.getSentDocumentsStoreKey(TestingSender), cdc.MustMarshalBinaryBare(&documentsIds))
+	store.Set(k.getReceivedDocumentsStoreKey(TestingRecipient), cdc.MustMarshalBinaryBare(&documentsIds))
+
+	newDocument := types.Document{
+		Uuid:       TestingDocument.Uuid + "new",
+		ContentUri: TestingDocument.ContentUri,
+		Metadata:   TestingDocument.Metadata,
+		Checksum:   TestingDocument.Checksum,
+	}
+	err := k.ShareDocument(ctx, TestingSender, ctypes.Addresses{TestingRecipient}, newDocument)
+	assert.Nil(t, err)
+
+	sentDocsBz := store.Get(k.getSentDocumentsStoreKey(TestingSender))
+	receivedDocsBz := store.Get(k.getReceivedDocumentsStoreKey(TestingRecipient))
+
+	var sentDocs, receivedDocs types.DocumentIds
+	cdc.MustUnmarshalBinaryBare(sentDocsBz, &sentDocs)
+	cdc.MustUnmarshalBinaryBare(receivedDocsBz, &receivedDocs)
+
+	assert.Equal(t, 2, len(sentDocs))
+	assert.Contains(t, sentDocs, TestingDocument.Uuid)
+
+	assert.Equal(t, 2, len(receivedDocs))
+	assert.Contains(t, receivedDocs, TestingDocument.Uuid)
 }
 
 func TestKeeper_GetUserReceivedDocuments_EmptyList(t *testing.T) {
 	_, ctx, k := SetupTestInput()
 	store := ctx.KVStore(k.StoreKey)
-	store.Delete([]byte(types.ReceivedDocumentsPrefix + TestingRecipient.String()))
+	store.Delete(k.getReceivedDocumentsStoreKey(TestingRecipient))
```

Comment:
you shouldn't need this if you keep redeclaring store in every test with new `ctx` and `k`